src/Microsoft.ML.Api/TypedCursor.cs

```diff
@@ -302,53 +329,108 @@ private Action<TRow> GenerateSetter(IRow input, int index, InternalSchemaDefinit
                         // DvText -> String
                         Ch.Assert(colType.IsText);
                         Ch.Assert(peek == null);
-                        return CreateTextToStringSetter(input, index, poke);
+                        return CreateActionSetter<DvText, string>(input, index, poke, (x) => x.ToString());
                     }
                     else if (fieldType == typeof(bool))
                     {
                         Ch.Assert(colType.IsBool);
                         Ch.Assert(peek == null);
-                        return CreateDvBoolToBoolSetter(input, index, poke);
+                        return CreateActionSetter<DvBool, bool>(input, index, poke, (x) => Convert.ToBoolean(x.RawValue));
                     }
-                    else
+                    else if (fieldType == typeof(bool?))
                     {
-                        // T -> T
-                        Ch.Assert(colType.RawType == fieldType);
-                        del = CreateDirectSetter<int>;
+                        Ch.Assert(colType.IsBool);
+                        Ch.Assert(peek == null);
+                        return CreateActionSetter<DvBool, bool?>(input, index, poke, (x) => x.IsNA ? (bool?)null : (x.IsTrue ? true : false));
```

Comment:
>x.IsNA ? (bool?)null : (x.IsTrue ? true : false) [](start = 92, length = 48)

We have this here:

[https://github.com/dotnet/machinelearning/blob/839bd6d96cc8afc946defed39f5ddb2028cf852a/src/Microsoft.ML.Core/Data/DvBool.cs#L110](https://github.com/dotnet/machinelearning/blob/839bd6d96cc8afc946defed39f5ddb2028cf852a/src/Microsoft.ML.Core/Data/DvBool.cs#L110)

So would a straightforward `x => (bool?)x` not work, rather than reproducing the logic in this method?